src/state/workspaces/slice.ts

```diff
@@ -18,11 +24,11 @@ export const workspacesSlice = createSlice({
 		setWorkspaces: (state, action: PayloadAction<WorkspaceMetadata[]>) => {
 			state.list = action.payload;
 		},
-		setWorkspace: (state, action: PayloadAction<WorkspaceMetadata | undefined>) => {
-			state.selected = action.payload;
-			applicationDataManager.setWorkspace(action.payload?.fileName, action.payload?.name);
+		setSelectedWorkspace: (state, action: PayloadAction<WorkspaceMetadata | undefined>) => {
+			const workspace = action.payload;
```

Comment:
I think this might be left over from debugging, but could just set state.selected to action.payload directly.